src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/IntegerCopyNumberTransitionMatrixCollection.java

```diff
@@ -0,0 +1,167 @@
+        maxCopyNumberInCollection = -1;
+    }
+
+    /**
+     *
+     * @param sexGenotype
+     * @param contig
+     * @param priorData
+     */
+    public void add(@Nonnull final String sexGenotype,
+                    @Nonnull final String contig,
+                    @Nonnull final IntegerCopyNumberTransitionMatrixData priorData) {
+        final SexGenotypeContigPairKey key = SexGenotypeContigPairKey.of(sexGenotype, contig);
+        if (sexGenotypeAndContigToTransitionMatrixMap.keySet().contains(key)) {
+            throw new IllegalArgumentException(String.format("The integer copy number transition matrix for sex" +
+                    " genotype \"%s\" and contig \"%s\" is already defined. Check for duplicate entries in the" +
+                    " table file.", sexGenotype, contig));
+        }
+        sexGenotypeAndContigToTransitionMatrixMap.put(key, priorData);
+        maxCopyNumberInCollection = FastMath.max(maxCopyNumberInCollection, priorData.getMaxCopyNumber());
+    }
+
+    /**
+     *
+     * @param sexGenotype
+     * @param contig
+     * @return
+     */
+    public IntegerCopyNumberTransitionMatrixData get(@Nonnull final String sexGenotype,
+                                                     @Nonnull final String contig) {
+        return sexGenotypeAndContigToTransitionMatrixMap.get(checkKey(sexGenotype, contig));
+    }
+
+    /**
+     * Returns a set of sex genotypes that are annotated at least once
+     *
+     * @return
+     */
+    public Set<String> getSexGenotypes() {
+        return sexGenotypeAndContigToTransitionMatrixMap.keySet().stream()
+                .map(SexGenotypeContigPairKey::getSexGenotype)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Returns a set of contigs that are annotated at least once
+     *
+     * @return
+     */
+    public Set<String> getContigs() {
+        return sexGenotypeAndContigToTransitionMatrixMap.keySet().stream()
+                .map(SexGenotypeContigPairKey::getContig)
+                .collect(Collectors.toSet());
+    }
+
+    public static IntegerCopyNumberTransitionMatrixCollection read(final File tableFile) {
+        return IntegerCopyNumberTransitionMatrixCollectionReader.read(tableFile);
+    }
+
+    public static IntegerCopyNumberTransitionMatrixCollection read(final Reader reader,
+                                                                   final String parentPath) {
+        return IntegerCopyNumberTransitionMatrixCollectionReader.read(reader, parentPath);
+    }
+
+    /**
+     * Asserts that all of provided sex genotypes and contigs have assigned copy number transition
+     * matrices.
+     *
+     * @param sexGenotypes a set of sex genotype identifier strings
+     * @param contigs      a set of contig strings
+     * @throws UserException.BadInput if a required data entry is not available
+     */
+    public void assertCompleteness(@Nonnull final Set<String> sexGenotypes,
```

Comment:
I can write a builder pattern to make the collection immutable but I don't see what's the gain except for aesthetics. We have a `add(...)` method safeguarded against duplicate entries, a `get(...)` method that throws exception if the key is not found, and an `assertCompleteness()` if the user cares to see if all pairwise (sex genotype, contig) keys have entries. It's a simple class with well-defined behavior. If you don't mind, I leave it as it is. It is impossible to make things truly immutable in java without expensive deep copying anyway...